andela/mazus-ah-backend
# #167165058 Reset Password feature

### What does this PR do?
User should be able to reset password

#### Description of Task to be completed?
wrote an integration test for forgot password and reset password
wrote forgot password and reset password controllers
adds a helper function to interface with the postal of email
adds a helper function to verify jw token
adds routes for both controllers

#### How should this be manually tested?

> 1. Fetch this branch on the terminal using 
> `git fetch origin ft-resetpassword-167165058`
> 2. Install dependencies using `npm install`
> 3. Start the server with `npm run server`
> 4. On Postman, sign up by making a POST request to /api/v1/auth/signup.
> 5. On Postman, make a POST request to /api/v1/auth/forgotpassword.
> 5. Check your inbox of the email you signed up with for a reset password link from authors haven.
> 6. Upon redirection to the browser, copy the link in the URL bar.
> 7. Paste this link in Postman and make a PATCH request.
> 

#### Any background context you want to provide?
User cannot reset the password with the same token more than once

#### What are the relevant pivotal tracker stories?
[#167165058](https://www.pivotaltracker.com/story/show/167165058)

#### Screenshots (if appropriate)
